torch/csrc/autograd/custom_function.cpp

```diff
@@ -273,18 +273,28 @@ optional_variable_list _process_backward_mode_ad(
     const std::unordered_set<at::TensorImpl*>& non_differentiable,
     const std::unordered_set<at::TensorImpl*>& dirty_inputs,
     const at::ArrayRef<c10::optional<Variable>> raw_outputs,
-    const std::shared_ptr<Node>& cdata) {
+    const std::shared_ptr<Node>& cdata,
+    const std::unordered_set<at::TensorImpl*>& to_save_if_setup_context) {
   int num_outputs = raw_outputs.size();
 
+  std::string error_msg_input_returned_as_is =
```

Comment:
Are we sure the compiler optimizes this well? `const char*` would give me more confidence

Reply:
OK, will update to const char* to be safe